tests/api_fastapi/core_api/routes/public/test_dag_run.py

```diff
@@ -156,6 +164,275 @@ def test_get_dag_run_not_found(self, test_client):
+    def test_filters(self, test_client, dag_id, query_params, expected_dag_id_list):
+        response = test_client.get(f"/public/dags/{dag_id}/dagRuns", params=query_params)
+        assert response.status_code == 200
+        body = response.json()
+        assert [each["run_id"] for each in body["dag_runs"]] == expected_dag_id_list
+
+    def test_bad_filters(self, test_client):
+        query_params = {
+            "logical_date_gte": "invalid",
+            "start_date_gte": "invalid",
+            "end_date_gte": "invalid",
+            "logical_date_lte": "invalid",
+            "start_date_lte": "invalid",
+            "end_date_lte": "invalid",
+        }
+        expected_detail = [
+            {
+                "type": "datetime_from_date_parsing",
+                "loc": ["query", "logical_date_gte"],
+                "msg": "Input should be a valid datetime or date, input is too short",
+                "input": "invalid",
+                "ctx": {"error": "input is too short"},
+            },
+            {
+                "type": "datetime_from_date_parsing",
+                "loc": ["query", "logical_date_lte"],
+                "msg": "Input should be a valid datetime or date, input is too short",
+                "input": "invalid",
+                "ctx": {"error": "input is too short"},
+            },
+            {
+                "type": "datetime_from_date_parsing",
+                "loc": ["query", "start_date_gte"],
+                "msg": "Input should be a valid datetime or date, input is too short",
+                "input": "invalid",
+                "ctx": {"error": "input is too short"},
+            },
+            {
+                "type": "datetime_from_date_parsing",
+                "loc": ["query", "start_date_lte"],
+                "msg": "Input should be a valid datetime or date, input is too short",
+                "input": "invalid",
+                "ctx": {"error": "input is too short"},
+            },
+            {
+                "type": "datetime_from_date_parsing",
+                "loc": ["query", "end_date_gte"],
+                "msg": "Input should be a valid datetime or date, input is too short",
+                "input": "invalid",
+                "ctx": {"error": "input is too short"},
+            },
+            {
+                "type": "datetime_from_date_parsing",
+                "loc": ["query", "end_date_lte"],
+                "msg": "Input should be a valid datetime or date, input is too short",
+                "input": "invalid",
+                "ctx": {"error": "input is too short"},
+            },
+        ]
+        response = test_client.get(f"/public/dags/{DAG1_ID}/dagRuns", params=query_params)
+        assert response.status_code == 422
+        body = response.json()
+        assert body["detail"] == expected_detail
+
+    def test_invalid_state(self, test_client):
+        with pytest.raises(ValueError, match="'invalid' is not a valid DagRunState"):
```

Comment:
I checked this for List TI endpoint and its same. It raises a ValueError. So, I kept it the same. But, wondering if this should raise 422